src/compiler/scala/tools/nsc/transform/Delambdafy.scala

```diff
@@ -105,6 +107,12 @@ abstract class Delambdafy extends Transform with TypingTransformers with ast.Tre
             // ... or an invokedynamic call
             super.transform(apply)
         }
+      case Template(_, _, _) =>
+        try {
+          // during this call boxingBridgeMethods will be populated from the Function case
+          val Template(parents, self, body) = super.transform(tree)
+          Template(parents, self, body ++ boxingBridgeMethods)
```

Comment:
I couldn't easily/lazily convince myself this deals correctly with nested templates. Could you shed some light on that?